save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt

```diff
@@ -18,9 +19,15 @@ class ToolStorage(
 ) : AbstractFileBasedStorage<ToolKey>(Path.of(configProperties.fileStorage.location) / TOOLS_PATH) {
     @Suppress("DestructuringDeclarationWithTooManyEntries")
     override fun buildKey(rootDir: Path, pathToContent: Path): ToolKey {
-        val (_, ownerName, toolName, vcsTagName, executableName) = pathToContent.pathNamesTill(rootDir)
+        val (ownerName, toolName, vcsTagName, executableName) = pathToContent.pathNamesTill(rootDir)
         return ToolKey(ownerName, toolName, vcsTagName, executableName)
     }
 
     override fun buildPathToContent(rootDir: Path, key: ToolKey): Path = rootDir / key.ownerName / key.toolName / key.vcsTagName / key.executableName
+
+    override fun isKey(rootDir: Path, pathToContent: Path): Boolean = pathToContent.name != DS_STORE
+
+    companion object {
+        private const val DS_STORE = ".DS_Store"
```

Comment:
I suggest this is converted to a list, because there may be other ignore patterns like `*~` (backup files created by text editors) or `Thumbs.db` (Windows Explorer metadata) and renamed to smth like `IGNORE_PATTERNS`:

```suggestion
        private val IGNORE_PATTERNS = listOf(".DS_Store")
```

Reply:
Fixed it in a bit different way - improved AbstractFileBasedStorage to have `storageIgnore` (by analogy with `gitignore`) - list of filenames that should not be considered as files from storage